server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,213 @@
+# we look for.
+find ${results_dir} -maxdepth 3 -type l -name '*.tar.xz' > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read tblink ;do
+    # resolve the link
+    tb=$(readlink -f ${tblink})
+    sts=$?
+    if [ $sts != 0 ] ;then
+        echo "$TS: readlink failed: ${tblink}" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    dest=$archive/$controller
+
+    pushd ${tbdir} >/dev/null 2>&1
+    prefix=""
+    # if necessary, fix prefix file name.
+    if [ -f "prefix.${resultname}" ] ;then
+        # rename it - using a prefix form was a mistake.
+        mv prefix.${resultname} ${resultname}.prefix
+    fi
+    # ... and remember it
+    if [ -f "${resultname}.prefix" ] ;then
+        prefix="${tbdir}/${resultname}.prefix"
+    fi
+    popd >/dev/null 2>&1
+
+    # sanity checks
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" | tee -a $status >&4
+        if [ -f ${tb}.md5 ] ;then
+            quarantine ${quarantine}/${controller} ${tb}.md5 ${prefix}
+        fi
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    if [ ! -f ${tb}.md5 ] ;then
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        quarantine ${quarantine}/${controller} ${tb} ${prefix}
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # version 001 agents check for duplicates in pbench-move-results but the check
+    # is almost completely ineffective now, since they can't see the "real" archive
+    # directory where the tarballs are stored. They will only be able to see any
+    # duplicates in the reception area and only during the short time before this
+    # script gets to them and moves them.
+    if [ -f ${dest}/${resultname}.tar.xz -o -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        echo "$TS: ${tb} duplicate name" >&4
+        quarantine $duplicates/$controller  ${tb} ${tb}.md5 $prefix
+        ndups=$ndups+1
+    fi
+
+    pushd ${tbdir} >/dev/null 2>&1
+    # MD5 check
+    md5sum --check $(basename $tb).md5
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: ${tb} failed MD5 check" >&4
+        quarantine ${quarantine}/${controller} ${tb} ${tb}.md5 ${prefix}
+        nquarantined=${nquarantined}+1
+        nerrs=${nerrs}+1
+        popd >/dev/null 2>&4
+        continue
+    fi
+    popd >/dev/null 2>&1
+
+    # make the destination directory and its TODO subdir if necessary.
+    mkdir -p ${dest}/TODO
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mkdir -p ${dest}\", status $sts" |
+            tee -a $status >&4
+        quarantine ${errors}/${controller} ${tb} ${tb}.md5 $prefix
```

Comment:
Should we count the number of times we quarantined data?

Reply:
This is counted as an error only. Basically, the (perhaps misnamed) function quarantine moves stuff to different destinations: quarantine, duplicates or errors.

Reply:
As we talked over the wall, perhaps have the counts add up properly so that we can tell what actions were taken by the report only?